cryptography/src/ed25519.rs

```diff
@@ -125,6 +125,47 @@ impl Scheme for Ed25519 {
     }
 }
 
+impl Ed25519 {
+    /// Check each signature is valid for the given message and public key.
+    ///
+    /// Because namespace is prepended to message before signing, the namespace
+    /// provided here must match the namespace provided during signing.
+    pub fn batch_verify(inputs: Vec<SignatureVerificationInput>) -> bool {
+        let mut batch = ed25519_consensus::batch::Verifier::new();
+        for input in inputs {
+            let item: ed25519_consensus::batch::Item = match input.try_into() {
+                Ok(item) => item,
+                Err(_) => return false,
+            };
+            batch.queue(item);
+        }
+        batch.verify(rand::thread_rng()).is_ok()
+    }
+}
+
+/// SignatureVerificationInput represents required information to verify a Signature.
+pub struct SignatureVerificationInput<'a> {
+    namespace: Option<&'a [u8]>,
+    message: &'a [u8],
+    public_key: &'a PublicKey,
+    signature: &'a Signature,
+}
+
+impl TryInto<ed25519_consensus::batch::Item> for SignatureVerificationInput<'_> {
+    type Error = ed25519_consensus::Error;
+    fn try_into(self) -> Result<ed25519_consensus::batch::Item, Self::Error> {
+        let public_key =
+            ed25519_consensus::VerificationKeyBytes::try_from(self.public_key.as_ref())?;
+        let sig = ed25519_consensus::Signature::try_from(self.signature.as_ref())?;
+        let payload = match self.namespace {
+            Some(namespace) => union_unique(namespace, self.message),
+            None => self.message.to_vec(),
```

Comment:
We should be able to avoid this `to_vec` conversion with the `CoW` approach I took in `bls12381` for handling namespace.